app/controllers/application_controller/performance.rb

```diff
@@ -1577,9 +1577,9 @@ def perf_remove_report_cols(report, charts = nil)
   # rpt.where_clause[2] should be id
   def perf_filter_charts(chart_layout, rpt)
     return unless rpt&.where_clause
-    model = rpt.where_clause[1].constantize
-    id = rpt.where_clause[2]
-    if model.find(id).type == "ManageIQ::Providers::Amazon::CloudManager::AvailabilityZone"
+    model = rpt.where_clause.dig(1).constantize
+    id = rpt.where_clause.dig(2)
+    if model.find(id).try(:type) == "ManageIQ::Providers::Amazon::CloudManager::AvailabilityZone"
```

Comment:
@PanSpagetka have you considered being more explicit in the condition, instead of using `try()`, to make sure the condition dosn't fail?

The `if` statement applies only for Amazon Availability zone, so I would avoid hacking the rest of the code, and would try to make sure the `model` is what you expect it to be if the model's record type is `ManageIQ::Providers::Amazon::CloudManager::AvailabilityZone` .

So maybe something like:

```ruby
if model == "AvailabilityZone" && model.find(id).try(:type) == "ManageIQ::Providers::Amazon::CloudManager::AvailabilityZone"
  ...
end
```

Anyway, the change you did works :+1:

Reply:
How does the `dig` help in case of a single argument?

Reply:
Dig returns nil instead of raising an exception. `a = [].dig(1) # a is now nil`